ibis/expr/types/temporal.py

```diff
@@ -578,7 +578,12 @@ class IntervalColumn(Column, IntervalValue):
 
 
 @public
-class DayOfWeek(Expr):
+class DayOfWeek:
+    """A namespace of methods for extracting day of week information."""
+
+    def __init__(self, expr):
+        self._expr = expr
```

Comment:
Could we generalize this? I think this kind of objects are going to be more frequest as the API evolves. There is similar class for the case API at https://github.com/ibis-project/ibis/blob/master/ibis/expr/builders.py

Reply:
What benefits would generalizing this bring? What is there to generalize?

Reply:
Nothing in this particular example. Just thinking loudly on how can we extend the API's functionality in the future.